### ISSUE TITLE
More over zealous route linking in markdown

### ISSUE THREAD

**Issue by brendanheywood** [theCrag/website#2492]
eg in:

https://www.thecrag.com/discussion/1017211590/bouldering-around-armidale#m1036433202

![image](https://cloud.githubusercontent.com/assets/187449/21490294/e6dcc29c-cc46-11e6-88a6-94606c8f89d6.png)

This is linking to a route in a completely different crag. I'm not sure exactly what the current business rules are but I think this is too much. Either we don't link for single word route names, or we just make it simple and require quotes. I don't think simply blacklisting certain names will be that productive for this bug.

**Comment by scd** [comment]
Can we not link if a single word and at the beginning of a sentence. 

We should probably hold off making changes until an language implementations has come through. I am guessing that with the use of upper case in some languages (eg German use of upper case in nouns) we will have to force quotes.

**Comment by brendanheywood** [comment]
Yeah my gut says the more workarounds we add the more edge cases we will create, I think just forcing quotes depending on the context (eg in discussions) is simple and understandable.

**Comment by brendanheywood** [comment]
Another example with a route called 'What':
![image](https://user-images.githubusercontent.com/187449/40116851-584a56d4-5958-11e8-8792-390e131339ee.png)
https://www.thecrag.com/discussion/1799997894/so-the-grade-25-trad-line-stophs-proj-is-still-an-unclimbed-project-and-not-to-be-tried-by-others#m1802514711

https://www.thecrag.com/climbing/australia/blue-mountains/mount-piddington/route/15380833

As we go more multi lingual this will get trickier, so that strengthens the forced quotes idea in certain contexts (if not all?)

**Comment by brendanheywood** [comment]
This one is even weirder:

https://www.thecrag.com/discussion/1812000417/australia--aussie-version-of-alex-honnold

This thread is in australia.

'Matt' is a route in Kalymnos: https://www.thecrag.com/climbing/greece/kalymnos/route/241111167
'Everyone' is a route in Korea: https://www.thecrag.com/climbing/south-korea/route/946916553

![image](https://user-images.githubusercontent.com/187449/40403544-de4330fc-5e95-11e8-976b-da6feef3814d.png)

Why on earth are they linking - but also if they are linking then it must be doing a horrendously expensive scan of basically every route name in the db to find potential matches.

**Comment by brendanheywood** [comment]
https://www.thecrag.com/discussion/1813267536/bells-line-of-road--does-anyone-see-any-issue-with-moving-bung-crag-and-dingo-creek-to-the-wolgan-valley-section-there#m1813284507

Discussion in the blue mountains linked the word 'NEW' to a route in tasmania

https://www.thecrag.com/discussion/1813267536/bells-line-of-road--does-anyone-see-any-issue-with-moving-bung-crag-and-dingo-creek-to-the-wolgan-valley-section-there#m1813288239

and the word IF to a route in Frog

There again are outside the scope of this discussion so not sure what is going on, but there are also a bunch of other auto links to routes inside the blue mountains which probably also don't make sense either

**Comment by brendanheywood** [comment]
Oh wow this one is pretty bad as it has auto linked the word 'Climbing' which is a gym route:

https://www.thecrag.com/discussion/1812000417/aussie-version-of-alex-honnold#m1812447957

https://www.thecrag.com/climbing/australia/queensland/brisbane/rocksports-indoor-climbing/foyer/rope7/route/1119823746

**Comment by brendanheywood** [comment]
Another example of an auto linked name, but from an outside forum conversation and linking into a gym route name which I think should never happen:

https://www.thecrag.com/discussion/2115240993/what-boulder-problem-is-this#m2115587520

**Comment by brendanheywood** [comment]
From @gogulfresh via https://github.com/theCrag/website/issues/3314

**What happened?**
A link to a route is created where it shouldn't/I don't want. How can linking be suppressed?

Example url(s) to reproduce the problem:
https://www.thecrag.com/climbing/austria/route/1529281464 
The word "spalt" in the description links to the route "spalt" 
https://www.thecrag.com/climbing/austria/route/1532385009
spalt ~= crack in german, thus in the text it is refering to a crack of this route and not the route "spalt"

Ideally also provide screenshots:


**What you expected:**

* [ ] Have a option to suppress linking for a certain word
* [ ] Have it documented here https://www.thecrag.com/en/article/updatingdescriptions

**Comment by rouletout** [comment]
one more: https://www.thecrag.com/climbing/switzerland/mittelland/area/2111171118
![screen shot 2018-11-21 at 17 35 51](https://user-images.githubusercontent.com/19162952/48925191-9e14ed00-ee86-11e8-8581-7bac852a289a.png)

**Comment by gogulfresh** [comment]
Another one: https://www.thecrag.com/climbing/austria/route/1880663691
<img width="752" alt="bildschirmfoto 2018-11-23 um 15 58 38" src="https://user-images.githubusercontent.com/18306619/48949624-e260c580-ef38-11e8-9a6b-6200b1c12815.png">

Verschneidung == book/corner/dihedral (What is the correct translation?)

**Comment by rouletout** [comment]
Another one:

https://www.thecrag.com/discussion/2203101762/kletterwerk--neue-boulderrouten

<img width="625" alt="screen shot 2019-01-16 at 10 13 19" src="https://user-images.githubusercontent.com/19162952/51296515-270d5800-19e2-11e9-88be-3761490523fb.png">

**Comment by rouletout** [comment]
Another one:

https://www.thecrag.com/discussion/2254853829

<img width="673" alt="screen shot 2019-02-13 at 10 09 57" src="https://user-images.githubusercontent.com/19162952/52725840-9477be80-2f77-11e9-8267-a331165a1a6c.png">

**Comment by rouletout** [comment]
Another one:

https://www.thecrag.com/discussion/2304627114/australia--mountainfilm-on-tour

<img width="612" alt="Screen Shot 2019-03-13 at 08 32 33" src="https://user-images.githubusercontent.com/19162952/54287226-e3ab1200-456a-11e9-8edb-6dc16d55e25b.png">

**Comment by brendanheywood** [comment]
Ugghh so many duds in a single thread, and picked up randomly from crags all over the world:

* [ ] 'The crag' Norway https://www.thecrag.com/discussion/2442620526/australia--can-one-redpoint-a-climb-twice 
* [ ] 'In' USA https://www.thecrag.com/discussion/2442620526/can-one-redpoint-a-climb-twice#m2443350120
* [ ] 'look' Romania https://www.thecrag.com/discussion/2442620526/can-one-redpoint-a-climb-twice#m2444704902
* [ ] 'at' Aus https://www.thecrag.com/discussion/2442620526/can-one-redpoint-a-climb-twice#m2445078111
* [ ] 'Otherwise' USA https://www.thecrag.com/discussion/2442620526/can-one-redpoint-a-climb-twice#m2445078111
* [ ] 'All free' France https://www.thecrag.com/discussion/2442620526/can-one-redpoint-a-climb-twice#m2446304130
* [ ] 'attempt' Lesotho https://www.thecrag.com/discussion/2442620526/can-one-redpoint-a-climb-twice#m2446304130

I got the grumps with this and dug around a little, and it turns out we always went up a node and then searched inside that, so if the discussion was at the australia level then we searched the entire world looking for matches. I've refined this a little, so it only searchs the TLC, or "this" node, or a routes parent cliff and added an explicit exclusion for the world. I think this is about 60% of the solution. I still think even within a crag there are too may false matches but at least they are not now from random crags around the world.

**Comment by brendanheywood** [comment]
* [ ] https://www.thecrag.com/discussion/2558717082/climbing-in-the-blue-mountains-in-january#m2559071514 A route 'What':

![image](https://user-images.githubusercontent.com/187449/59551131-3d7f0600-8fb8-11e9-8333-b06d11068e2b.png)


https://www.thecrag.com/climbing/australia/blue-mountains/mount-piddington/route/15380833

**Comment by georg-d** [comment]
> From @gogulfresh via #3314
...
>     * [ ]  Have it documented here https://www.thecrag.com/en/article/updatingdescriptions

Please do not add the details there, but add only a link to the (still to be done) description in https://www.thecrag.com/en/article/formattingtext#internal-links because the links do work in many other text fields than route descriptions.

**Comment by rouletout** [comment]
Thanks for all the comments. This "feature" has become more of a pain than anything else and we are considering to remove it completely. the idea is to require hyphens or simialr to create the link but remove all links that are strictly text based as in the examples above.

If you have any commenst on this, please put them here.
